src/main/java/org/zalando/nakadi/service/EventTypeService.java

```diff
@@ -281,6 +257,29 @@ public void delete(final String eventTypeName, final Client client)
         }
     }
 
+    private void updateRetentionTime(final EventType original, final EventType eventType) throws NakadiException {
+        final Long newRetentionTime = eventType.getOptions().getRetentionTime();
+        final Long oldRetentionTime = original.getOptions().getRetentionTime();
+        if (oldRetentionTime == null) {
+            // since we have some inconsistency in DB I will put here for a while
+            throw new InconsistentStateException("Empty value for retention time in existing EventType");
+        }
+        boolean retentionTimeUpdated = false;
+        try {
+            if (newRetentionTime != null && !newRetentionTime.equals(oldRetentionTime)) {
+                updateTopicRetentionTime(original.getName(), newRetentionTime);
+            } else {
+                eventType.setOptions(original.getOptions());
+            }
+            updateEventTypeInDB(eventType, newRetentionTime, oldRetentionTime);
```

Comment:
The point is to segregate this logic from the update method, because it is quite big and has a lot of different functionality. I would just rename it as well as updateEventTypeInDB, because it does update of timeline in addition to update of et.